src/transaction/requests.rs

```diff
@@ -125,13 +125,20 @@ pub fn new_scan_request(
     timestamp: u64,
     limit: u32,
     key_only: bool,
+    reverse: bool,
 ) -> kvrpcpb::ScanRequest {
     let mut req = kvrpcpb::ScanRequest::default();
-    req.set_start_key(start_key);
-    req.set_end_key(end_key);
+    if !reverse {
+        req.set_start_key(start_key);
```

Comment:
This violates the request spec in TiKV, we assume the start key is always no larger than the end key.

Reply:
You can check the client-go implementation: https://github.com/tikv/client-go/blob/bb026bdad8f34604cc01fdafbaf4063729583a44/txnkv/txnsnapshot/scan.go#L194

Reply:
@iosmanthus I check the proto definition in kvrpcpb.proto https://github.com/tikv/client-rust/blob/master/tikv-client-proto/proto/kvrpcpb.proto#L57 

```
    // For compatibility, when scanning forward, the range to scan is [start_key, end_key), where start_key < end_key;
    // and when scanning backward, it scans [end_key, start_key) in descending order, where end_key < start_key.
```

The start_key and end_key is define as `bound range` , (start_key < end_key is guaranteed).

If reverse scan request be created, we use the range [req.end_key, req.start_key) to create scan request which should be initialized use 

```
        req.set_start_key(end_key);
        req.set_end_key(start_key);
```
which  guarantee the req.end_key<req.start_key.

Reply:
And the integration test works as expected.